pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/ItemsRule.scala

```diff
@@ -79,51 +105,45 @@ object ItemsRule extends FieldMergeRule with MergerLogging {
              sources: NonEmptyList[TransformedBaseWork]): FieldData = {
       val sierraItems = target.data.items
       val metsItems = sources.toList.flatMap(_.data.items)
-      val metsUrls = metsItems.flatMap(_.locations).collect {
-        case DigitalLocation(url, _, _, _, _, _) => url
-      }
+
       debug(s"Merging METS items from ${describeWorks(sources)}")
       sierraItems match {
         case List(sierraItem) =>
           List(
             sierraItem.copy(
-              locations = sierraItem.locations.filterNot(hasUrl(metsUrls)) ++
-                metsItems.flatMap(_.locations)
+              locations = sierraItem.locations ++ metsItems.flatMap(_.locations)
             )
           )
-        case _ =>
-          sierraItems.filterNot(_.locations.exists(hasUrl(metsUrls))) ++
-            metsItems
+        case multipleItems =>
+          multipleItems ++ metsItems
       }
     }
   }
 
-  private val mergeMiroPhysicalAndDigitalItems = new PartialRule {
-    val isDefinedForTarget: WorkPredicate = WorkPredicates.sierraWork
-    val isDefinedForSource
-      : WorkPredicate = WorkPredicates.miroWork or WorkPredicates.digitalSierra
+  /** We merge the Miro location to the Sierra work with a single item.
+    * If there are multiple items, we don't merge.
+    */
+  private val mergeMiroItems = new PartialRule {
+    val isDefinedForTarget: WorkPredicate = WorkPredicates.singleItemSierra
+    val isDefinedForSource: WorkPredicate = WorkPredicates.miroWork
 
     def rule(target: UnidentifiedWork,
-             sources: NonEmptyList[TransformedBaseWork]): FieldData =
-      (target.data.items, sources.toList.partition(WorkPredicates.miroWork)) match {
-        case (List(sierraSingleItem), (miroSources, sierraSources)) =>
+             sources: NonEmptyList[TransformedBaseWork]): FieldData = {
+      val sierraItems = target.data.items
+      val miroItems = sources.toList.flatMap(_.data.items)
+
+      debug(s"Merging Miro items from ${describeWorks(sources)}")
+      sierraItems match {
+        case List(sierraItem) =>
           List(
-            sierraSingleItem.copy(
-              locations = sierraSingleItem.locations ++
-                (sierraSources ++ miroSources).flatMap(
-                  _.data.items.flatMap(_.locations)
-                )
+            sierraItem.copy(
+              locations = sierraItem.locations ++ miroItems.flatMap(_.locations)
             )
           )
-        case (multipleSierraItems, (_, sierraSources)) =>
-          multipleSierraItems ++ sierraSources.flatMap(_.data.items)
+        // This case is unreachable due to the `singleItemSierra` predicate
```

Comment:
Maybe it would be clearer to replace this block with a map in this case? Not sure, removes the redundant arm of the pattern match at least, but doesn't represent that there's only ever one item.
```
sierraItems.map { sierraItem =>
  sierraItem.copy(
    locations = sierraItem.locations ++ miroItems.flatMap(_.locations)
  )
}
```

Reply:
This wouldn't work as we only want to add the locations if there is a single sierra item, hence the pattern match.

Reply:
But we've already restricted the sierra item to be a `singleItemSierra`

Reply:
I've done it using head to be more explicit about what we're doing. Is that nicer?

Reply:
Yeah that works too - although to be truly nitpicky we could also use `headOption.map{...}.getOrElse(...)` to stay type safe...

Reply:
I prefer that but we're not using it elsewhere.

And this makes it even more safe:
https://github.com/wellcomecollection/catalogue/blob/master/pipeline/merger/src/test/scala/uk/ac/wellcome/platform/merger/rules/WorkPredicatesTest.scala#L72-L77

For legibility I might keep it. And this will throw if something is up which might be better than swallowing it in a `getOrElse`.

You could have a `getOrElse(Left(PredicateFailedMe))` - but that would be a whole other refactor.